Sources/PackageModel/Target.swift

```diff
@@ -11,12 +11,53 @@
 import Basic
 
 public class Target: ObjectIdentifierProtocol {
+
+    /// The type of library.
+    public enum LibraryType: String, Codable {
+
+        /// Static library.
+        case `static`
+
+        /// Dynamic library.
+        case `dynamic`
+
+        /// The type of library is unspecified and should be decided by package manager.
+        case automatic
+    }
+
     /// The target kind.
-    public enum Kind: String {
+    public enum Kind: CustomStringConvertible, Equatable {
         case executable
-        case library
-        case systemModule = "system-target"
+        case library(LibraryType)
+        case systemModule
         case test
+        public var description: String {
```

Comment:
`description` is not intended for stable results i.e. those used in serialization formats. Also, the documentation explicitly states:

    Calling this property directly is discouraged.

Please return to `rawValue`.

Reply:
When I made the change, the swift compiler complained that you can't have a `rawValue` in conjunction with an enum with cases. In any case, this code was copied verbatim from `ProductType` in https://github.com/apple/swift-package-manager/blob/master/Sources/PackageModel/Product.swift#L17-L53

Reply:
I didn't mean the automatically synthesized implementation - you can define `rawValue` yourself (or call it something else).